### PR TITLE
Copy release

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/14156

This PR ensures the `release.giantswarm.io/version` label is set for `MachinePool` and `AzureMachinePool` CRs at creation time if not set.
It takes the value from the `Cluster` CR.

### CODE REVIEW

**Comment by tuommaki** [comment]
Hmm... Now that I'm thinking, is this necessarily needed at all? Theoretically in NP-to-NP upgrade PR, the `clusterupgrade` handler should always propagate latest release & azure-operator version labels to MachinePools that belong to that given cluster. `machinepoolupgrade` handler would then enforce the same propagation from `MachinePool` to corresponding `AzureMachinePool`. Maybe we should test this use-case with the the branch once these few other ones are done?

**Comment by whites11** [comment]
> Hmm... Now that I'm thinking, is this necessarily needed at all? Theoretically in NP-to-NP upgrade PR, the `clusterupgrade` handler should always propagate latest release & azure-operator version labels to MachinePools that belong to that given cluster. `machinepoolupgrade` handler would then enforce the same propagation from `MachinePool` to corresponding `AzureMachinePool`. Maybe we should test this use-case with the the branch once these few other ones are done?

Should it work on node pool creation as well?

**Comment by tuommaki** [comment]
> Should it work on node pool creation as well or during an upgrade only?

The Cluster -> MachinePool logic I need to carefully verify, but the MachinePool -> AzureMachinePool is unconditional, it should always work.

**Comment by whites11** [comment]
> > Should it work on node pool creation as well or during an upgrade only?
> 
> The Cluster -> MachinePool logic I need to carefully verify, but the MachinePool -> AzureMachinePool is unconditional, it should always work.

ok I will test it now.

**Comment by tuommaki** [comment]
> ok I will test it now.

You need to use the version from branch `machinepool-add-upgrading-condition` for that. There is release on `ghost`: `v42.0.1`.

I checked the code and when compiling/running it in my head, it works on creation as well.

**Comment by whites11** [comment]
> > ok I will test it now.
> 
> You need to use the version from branch `machinepool-add-upgrading-condition` for that. There is release on `ghost`: `v42.0.1`.
> 
> I checked the code and when compiling/running it in my head, it works on creation as well.

Tuomas the human compiler

**Comment by whites11** [comment]
In general Tuomas I don't know if I like the idea of leaving the population of labels to Azure Operator.
We used to set all those labels "in advance" and we should keep doing like that.
I feel like we could one day remove or change the label duplication feature from azure operator for whatever reason and the cluster creation will stop working.
This change here is reasonably simple and it solves the problem once and for all simply by keeping things as they were.

WDYT?

File: pkg/generic/ensure_release_label_test.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	}
+
+	if release != nil {
+		obj.Labels[label.ReleaseVersion] = *release
+	}
+
+	return obj
+}
+
+//func Test_WhenCreatingClusterWithExistingOrganizationThenValidationSucceeds(t *testing.T) {
```
Comment (tuommaki) [inline pkg/generic/ensure_release_label_test.go:145]: If this is not needed, please delete it so it doesn't confuse anyone in the future :slightly_smiling_face:

**Review by tuommaki — APPROVED** [review]
I'm fine either way. I was just bringing that up in case we want to keep the admission controller as small as possible, but this is straightforward in general so I'm fine with this. Do as you see best :+1:

**Comment by whites11** [comment]
BTW I tested it with the az op version you mentioned and worker nodes never came up.

**Pull request merged**